File: examples/fastapi/src/watcher.py
# watcher.py — Watcher24 client singleton for the FastAPI example.
# Import `client` from this module in routers and background tasks.
# One instance per process — the background flusher thread runs for the
# lifetime of the app and is stopped cleanly in the lifespan shutdown hook.
#
# app_id is optional when using an app-scoped key created via Settings → Apps
# in the console. The gateway resolves it automatically from the key.
# Only set W24_APP_ID if using a legacy org-level key with no app linked.
import os
from watcher_sdk import Client

_app_id = os.environ.get("W24_APP_ID") or None

client = Client(
    api_key=os.environ.get("W24_API_KEY", ""),
    app_id=_app_id,
    # service_name labels this component in the dashboard — filter by "fastapi-example"
    # to isolate these server events from other services in the same app.
    service_name=os.environ.get("W24_SERVICE_NAME", "fastapi-example"),
    # gateway_url defaults to https://ingest.watcher24.io — only set W24_GATEWAY_URL
    # for local dev (http://localhost:8080) or self-hosted deployments.
    gateway_url=os.environ.get("W24_GATEWAY_URL"),
    environment=os.environ.get("W24_ENVIRONMENT", "development"),
)

File: examples/nextjs/src/lib/watcher.ts
// watcher.ts — server-side Watcher24 client singleton.
// Import this in Server Components, Route Handlers, and Server Actions.
// Never import it in Client Components — use @watcher/nextjs/client hooks there.
//
// The globalThis cache prevents re-creating the client on every hot-reload in
// development, which would spin up many background flushers.
import { createNextServerClient } from "@watcher/nextjs/server";

declare global {
  // eslint-disable-next-line no-var
  var __watcher: ReturnType<typeof createNextServerClient> | undefined;
}

export const watcher =
  globalThis.__watcher ??
  (globalThis.__watcher = createNextServerClient({
    apiKey: process.env.W24_API_KEY ?? "",
    // appId links events to the registered application in the Watcher24 dashboard.
    // Get the ID from Settings → Apps.
    appId: process.env.W24_APP_ID,
    // serviceName labels this component so you can filter by service in the dashboard.
    // Change this to reflect what this Next.js instance is (e.g. "marketing-site", "dashboard-api").
    serviceName: process.env.W24_SERVICE_NAME ?? "nextjs-example",
    // gatewayUrl defaults to https://ingest.watcher24.io — only override for
    // local dev (http://localhost:8080) or self-hosted deployments.
    gatewayUrl: process.env.W24_GATEWAY_URL,
    environment: process.env.NODE_ENV ?? "development",
  }));
